### DISCUSSION TITLE
cmp column gets too narrow when the element names are too long

### DISCUSSION THREAD

**Discussion by drzhivago420** [hrsh7th/nvim-cmp#1374] [Ideas]
This is most likely an issue with my configuration, but I have no knowledge of how to solve it and don't know anywhere else to ask. Any way to fix this?
![Screenshot_20221205_181204](https://user-images.githubusercontent.com/110633694/205700171-82c24b4d-8bfa-4bbb-832d-3e006ae32aa5.png)

**Comment by hrsh7th** [comment]
What do you expect the fixed behavior?

Possible solutions are

1. Trim completion menu text
2. Specify minimam width for documentation window
    - And if the documentation window has no enough space, we avoid to show it.

**Reply by MaienM** [reply]
In the provided image there is not enough space to the sides, but there is enough space below the completion menu for a reasonably sized documentation window. Would it be possible to adjust the position of the documentation window in cases such as these? Either just always going for the side that has the most space available, or perhaps some heuristic like "if less than X columns available to the side, go to the top/bottom"?

**Comment by niveK77pur** [comment]
Hello, I have recently encountered the same issue and was also wondering if it weren't possible to *dynamically* move the documentation window to wherever there is more space, as suggested by @MaienM. But as mentioned by @drzhivago420, I could not find anything at a glance which could help me achieve/solve this; I am pretty much stuck in the same place now.

I have attached another example, where my terminal is too narrow for the docs to even fit on the right (if I make the window wider, and trigger the completion again, I get a similar looking result as @drzhivago420). As you can see, the docs end up occluding the completion suggestions and I cannot even see the suggestions anymore.

![image](https://user-images.githubusercontent.com/10981161/218728235-88c68208-554e-4826-915b-8235ed5d8f54.png)

It would indeed feel very intuitive to have the docs appear below the completion window in this case (or above, if my suggestion window happens to be put above the cursor).

@MaienM's proposition sounds like a neat way to dodge these cases in my eyes. In (very) extreme cases, one might take @hrsh7th's suggestion on either trimming the completion menu (as is currently done if we disregard the documentation window), or to avoid showing the docs altogether if it gets *reaallllly* cramped.

**Comment by eyebrowkang** [comment]
I have the same problem here, I think @hrsh7th's second suggestion is acceptable for me. I think documentation is not necessary for me (if it is too narrow, hiding is better), and I want my whole layout is beautiful, not odd-shaped.

**Comment by waiting-for-dev** [comment]
Same problem with elixir-lsp. What about being able to provide a `max_width` for the completion window with a horizontal scroll if needed?

![Screenshot from 2023-11-18 08-06-20](https://github.com/hrsh7th/nvim-cmp/assets/52650/a7dac1fd-9024-4cd5-89cb-7cf07ed93b0c)
